plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConnectorModule.java

```diff
@@ -78,15 +78,14 @@ public BigQueryClient provideBigQueryClient(BigQueryConfig config, HeaderProvide
     static String calculateBillingProjectId(Optional<String> configParentProjectId, Optional<Credentials> credentials)
     {
         // 1. Get from configuration
-        if (configParentProjectId.isPresent()) {
-            return configParentProjectId.get();
-        }
-        // 2. Get from the provided credentials, but only ServiceAccountCredentials contains the project id.
-        // All other credentials types (User, AppEngine, GCE, CloudShell, etc.) take it from the environment
-        if (credentials.isPresent() && credentials.get() instanceof ServiceAccountCredentials) {
-            return ((ServiceAccountCredentials) credentials.get()).getProjectId();
-        }
-        // 3. No configuration was provided, so get the default from the environment
-        return BigQueryOptions.getDefaultProjectId();
+        return configParentProjectId
+                // 2. Get from the provided credentials, but only ServiceAccountCredentials contains the project id.
+                // All other credentials types (User, AppEngine, GCE, CloudShell, etc.) take it from the environment
+                .orElseGet(() -> credentials
+                        .filter(cr -> cr instanceof ServiceAccountCredentials)
+                        .map(cr -> (ServiceAccountCredentials) cr)
```

Comment:
`ServiceAccountCredentials.class::cast`

Reply:
fixed in amended commit: 1005576e26b052427de07609d1cc7270fcfbd3b8